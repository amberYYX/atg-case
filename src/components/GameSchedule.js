import React from 'react'

const GameSchedule = ({ data, handleSelectedId }) => {
  const changeDateFormat = stringDate => {
    const date = new Date(stringDate)
    const time = stringDate.split('T')[1]
    const formatedDate = date.toDateString() + ', ' + time
    return formatedDate
  }

  const handleOnClick = item => {
    handleSelectedId(item)
  }

  return (
    <table className=" m-auto w-3/4 mt-8 bg-blue-600  text-white">
      <tbody>
        <tr className="text-center border-b border-blue-400 text-2xl">
          <td>Game Id</td>
          <td>Start Time</td>
        </tr>
        {data.map(item => (
          <tr
            key={item.id}
            className="text-center text-base border-b hover:bg-blue-800 leading-8 border-blue-500"
            onClick={() => handleOnClick(item)}
          >
            <td className="border-r border-blue-400">{item.id}</td>
            <td>{changeDateFormat(item.startTime)}</td>
          </tr>
        ))}
      </tbody>
    </table>
  )
}

export default GameSchedule
