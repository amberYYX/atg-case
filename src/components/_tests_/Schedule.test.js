import React from 'react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import GameSchedule from '../GameSchedule'

describe('schedule', () => {
  const testInfo = [
    {
      id: '001',
      startTime: '2020-11-16T12:20:00'
    },
    {
      id: '002',
      startTime: '2020-11-16T16:10:00'
    },
    {
      id: '003',
      startTime: '2020-11-16T17:10:00'
    }
  ]

  it('renders default correctly', () => {
    const tree = renderer.create(<GameSchedule data={testInfo} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders content correctly with game id & Formated start Time', () => {
    const component = render(<GameSchedule data={testInfo} />)
    expect(component.container).toHaveTextContent('001')
    expect(component.container).toHaveTextContent('Mon Nov 16 2020, 12:20:00')
  })
})
