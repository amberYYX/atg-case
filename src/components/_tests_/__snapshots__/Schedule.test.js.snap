// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schedule renders default correctly 1`] = `
<table
  className=" m-auto w-3/4 mt-8 bg-blue-600  text-white"
>
  <tbody>
    <tr
      className="text-center border-b border-blue-400 text-2xl"
    >
      <td>
        Game Id
      </td>
      <td>
        Start Time
      </td>
    </tr>
    <tr
      className="text-center text-base border-b hover:bg-blue-800 leading-8 border-blue-500"
      onClick={[Function]}
    >
      <td
        className="border-r border-blue-400"
      >
        001
      </td>
      <td>
        Mon Nov 16 2020, 12:20:00
      </td>
    </tr>
    <tr
      className="text-center text-base border-b hover:bg-blue-800 leading-8 border-blue-500"
      onClick={[Function]}
    >
      <td
        className="border-r border-blue-400"
      >
        002
      </td>
      <td>
        Mon Nov 16 2020, 16:10:00
      </td>
    </tr>
    <tr
      className="text-center text-base border-b hover:bg-blue-800 leading-8 border-blue-500"
      onClick={[Function]}
    >
      <td
        className="border-r border-blue-400"
      >
        003
      </td>
      <td>
        Mon Nov 16 2020, 17:10:00
      </td>
    </tr>
  </tbody>
</table>
`;
